src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Completion/VSCompletionList.cs

```diff
@@ -81,11 +83,18 @@ private static VSCompletionList PromoteCommonCommitCharactersOntoList(VSCompleti
             var promotedCompletionItems = new List<CompletionItem>();
             foreach (var completionItem in completionList.Items)
             {
-                if (completionItem.CommitCharacters == mostUsedCommitCharacters)
+                if (completionItem.CommitCharacters != null && SequenceEqual(completionItem.CommitCharacters, mostUsedCommitCharacters))
                 {
                     var clearedCompletionItem = completionItem with { CommitCharacters = null };
                     promotedCompletionItems.Add(clearedCompletionItem);
                 }
+                else if (completionItem is VSCompletionItem vsCompletionItem &&
```

Comment:
Should this be an `if` to protect in the misuse scenario where `CommitCharacter` & `VSCommitCharacters` accidentally both are non-null?